docs/modules/scylladb.md

```diff
@@ -0,0 +1,131 @@
+# ScyllaDB
+
+Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+## Introduction
+
+The Testcontainers module for ScyllaDB, a NoSQL database fully compatible with Apache Cassandra and DynamoDB, allows you
+to create a ScyllaDB container for testing purposes.
+
+## Adding this module to your project dependencies
+
+Please run the following command to add the ScyllaDB module to your Go dependencies:
+
+```shell
+go get github.com/testcontainers/testcontainers-go/modules/scylladb
+```
+
+## Usage example
+
+<!--codeinclude-->
+[Creating a ScyllaDB container](../../modules/scylladb/examples_test.go) inside_block:ExampleRun
+<!--/codeinclude-->
+
+## Module Reference
+
+### Run function
+
+- Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+The ScyllaDB module exposes one entrypoint function to create the ScyllaDB container, and this function receives three parameters:
+
+```golang
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*ScyllaDBContainer, error)
+```
+
+- `context.Context`, the Go context.
+- `string`, the Docker image to use.
+- `testcontainers.ContainerCustomizer`, a variadic argument for passing options.
+
+!!! info
+    By default, we add the `--developer-mode=1` flag to the ScyllaDB container to disable the various checks Scylla
+    performs.
+    Also In scenarios in which static partitioning is not desired - like mostly-idle cluster without hard latency
+    requirements, the --overprovisioned command-line option is recommended. This enables certain optimizations for ScyllaDB
+    to run efficiently in an overprovisioned environment. You can change it by using the `WithCustomCommand` function.
+
+### Container Options
+
+When starting the ScyllaDB container, you can pass options in a variadic way to configure it.
+
+#### Image
+
+If you need to set a different ScyllaDB Docker image, you can set a valid Docker image as the second argument in the
+`Run` function. Eg:
+
+```golang
+scylladb.Run(context.Background(), "scylladb/scylla:6.2.1")
+// OR
+scylladb.Run(context.Background(), "scylladb/scylla:5.6")
+```
+
+{% include "../features/common_functional_options.md" %}
+
+#### With Database Configuration File (scylla.yaml)
+
+- Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+In the case you have a custom config file for ScyllaDB, it's possible to copy that file into the container before it's
+started, using the `WithConfigFile(cfgPath string)` function.
```

Comment:
suggestion: let's use the `WithConfig(io.Reader)` method here, as suggested in the review comments. It's also applied to the existing tests, so we should change the method call in there.